src/exo6 - ReaderTaskEither/exo6.exercise.ts

```diff
@@ -82,22 +92,36 @@ export const getConcatenationOfTheTwoUserNamesUsingAp: (args: {
 // (the firstly fetched user) to perform a second operation (fetch their best friend)
 // and bind the return value to feed the context and use this data.
 
-export const getConcatenationOfTheBestFriendNameAndUserName: (args: {
-  userIdOne: string;
-}) => ReaderTaskEither<
-  User.Repository.Access,
-  User.Repository.UserNotFoundError,
-  string
-> = unimplemented;
+export const getConcatenationOfTheBestFriendNameAndUserName = ({userId}: {
+  userId: string;
+}) => pipe(
+  rte.Do,
+
+  rte.apS('user', User.Repository.getById(userId)),
+  rte.bindW('bestFriend', ({user}) => User.Repository.getById(user.bestFriendId)), 
+  rte.map(({user, bestFriend}) => capitalizeName(user.name) + capitalizeName(bestFriend.name)),
+
+  // rte.apS('user', User.Repository.getById(userId)),
+  // rte.apS('userName', getCapitalizedUserName({userId})),
+  // rte.bindW('bestFriendName', ({user}) => getCapitalizedUserName({userId: user.bestFriendId})), 
+  // rte.map(({userName, bestFriendName}) => userName + bestFriendName),
+
+  )
+
 
 // Most of the time, you will need to use several external services.
 // The challenge of this use-case is to use TimeService in the flow of our `rte`
-type Dependencies = User.Repository.Access & Application.TimeService.Access;
+// type Dependencies = User.Repository.Access & Application.TimeService.Access;
 
-export const getConcatenationOfUserNameAndCurrentYear: (args: {
+export const getConcatenationOfUserNameAndCurrentYear = ({userIdOne}: {
   userIdOne: string;
-}) => ReaderTaskEither<
-  Dependencies,
-  User.Repository.UserNotFoundError,
-  string
-> = unimplemented;
+}) => pipe (
+  rte.Do,
+  rte.apS('user', User.Repository.getById(userIdOne)),
+  rte.apSW('thisYear', rte.fromReader(Application.TimeService.thisYear())),  //added W because widening types to include number
+  rte.map(({user, thisYear}) => `${user.name}${thisYear}`)
+
+)
+
+// I checked the solution for the above exercise because I was unable to figure out
+// that "rte.fromReader" needed to proceed the code accessing the TimeService dependency.
```

Comment:
Let's discuss this in a call then in case you want some extra explanations about this?